src/CoreBundle/private/library/classes/TCMSFields/TCMSFieldExtendedLookup.class.php

```diff
@@ -152,6 +150,58 @@ protected function _GetOpenWindowJS(&$oPopupTableConf)
         return $js;
     }
 
+    private function getTargetListRestriction(): array
+    {
+        $restrictionExpression = $this->oDefinition->GetFieldtypeConfigKey('restriction');
+        if (null === $restrictionExpression) {
+            return [];
+        }
+        if (false === \strpos($restrictionExpression, '=')) {
+            return [];
+        }
+        $equalPosition = \mb_strpos($restrictionExpression, '=');
+        $restrictionField = \trim(\mb_substr($restrictionExpression, 0, $equalPosition));
+        $restrictionValue = \trim(\mb_substr($restrictionExpression, $equalPosition + 1));
+
+        if (\mb_strpos($restrictionValue, '[{') === 0 && '}]' === \mb_substr($restrictionValue, -2)) {
```

Comment:
Is space (trim()) a potential problem here?

(And possible simplification: if (true === $this->isTokenString()) and/or if ($this->startsWith() && $this->endsWith()).)

Reply:
> Is space (trim()) a potential problem here?

changed the code to keep the trim only for the `[{}]` variation (in which case the trim is relevant).

> 
> (And possible simplification: if (true === $this->isTokenString()) and/or if ($this->startsWith() && $this->endsWith()).)

replached with `isFieldToken` and moved trim into `isFieldToken` and `getFieldValueFromFieldToken`